### PR TITLE
Store settings in browser

### LINKED ISSUE #50: Store and load settings in browser
Use browser storage to store color pallets, rakes etc.

### PR DESCRIPTION
**Pull request by JBenda**
- opt-in setting (General) to enable setting storage
- stores every setting change AFTER enabling storage
- disabling storage deletes data

close #50

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -5,6 +5,27 @@
 		<meta name="description" content="A marebling tool">
 		<link rel="shortcut icon" href="favicon.ico" type="image/x-icon">
 		<link rel="stylesheet" href="stylesheet.css">
+		<script>
+			function switchStore() {
+				if(switchStore.input.checked) {
+					window.localStorage.setItem('store-active', '1');
+				} else {
+					if(window.confirm('When you the broswer storage all stored data will be cleard!\nthere is no way to undo this!')) {
```
Comment (t-erxleben) [inline frontend/index.html]: Maybe change to: When you disable browser storage all stored options will be cleared.\nThere is no way to undo this!

File: frontend/index.html [hunk 2]
```diff
@@ -24,6 +45,22 @@
...
 	<menu class="sidebar">
+		<!-- general -->
+		<li>
+				<input type="radio" name="sidebar" id="sidebar-general" />
+				<label for="sidebar-general">General</label>
+				<div>
+					<table>
+						<tr>
+							<td><label title="Enable to store settings in the browser. This will persist them even when you close and revisit the site later. This only effects settings which you changed after enableing storage!">store settings</label></td>
+							<td><label class="switch" title="Enable to store settings in the browser. This will persist them even when you close and revisit the site later. This only effects settings which you change afetr enableing storage!">
```
Comment (t-erxleben) [inline frontend/index.html]: pls fix typos :D

**Review by t-erxleben — CHANGES_REQUESTED** [review]
See comments.

File: frontend/index.html (hunk 1 above)
Comment (JBenda) [reply frontend/index.html:13]: But this is the reason why the user gets a dialog box to verify that he really wants to clear its data.
The ratio was that when you disable storage you don't want a clear setup and not you're old clutter :thinking:

**Comment by JBenda** [comment]
Hope it's now fixed?

**Review by t-erxleben — APPROVED** [review]
:+1:

**Pull request merged**